File: src/main/java/upbrella/be/store/repository/ClassificationRepository.java
package upbrella.be.store.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import upbrella.be.store.entity.Classification;
import upbrella.be.store.entity.ClassificationType;

import java.util.List;

public interface ClassificationRepository extends JpaRepository<Classification, Long> {

    List<Classification> findByType(ClassificationType type);
}

File: src/main/kotlin/upbrella/be/store/repository/ClassificationRepository.kt
package upbrella.be.store.repository

import org.springframework.data.jpa.repository.JpaRepository
import upbrella.be.store.entity.Classification
import upbrella.be.store.entity.ClassificationType

interface ClassificationRepository : JpaRepository<Classification, Long> {
    fun findByType(type: ClassificationType): List<Classification>
}